zerver/views/user_groups.py

```diff
@@ -237,3 +237,20 @@ def get_user_group_membership(
             "is_user_group_member": is_user_in_group(user_group, target_user, recursive=recursive)
         },
     )
+
+
+@has_request_variables
+def get_own_user_group_membership(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    user_group_id: int = REQ(json_validator=check_int, path_only=True),
+    recursive: bool = REQ(json_validator=check_bool, default=True),
```

Comment:
Do we want to name this parameter as something else like `include_subgroup_members` for the API or is this fine?

Reply:
Let's have an "#api design" discussion on this; I could imagine a name more like `include_subgroups`.

Actually, probably best to do one topic about each of the core changes here:
* API for this endpoint.
* Format of user group update events.
* Etc.